### ISSUE TITLE
StartKeywordRecognitionAsync StopsSession immediately after wake word without waiting 5 seconds for the user to speak

### ISSUE THREAD

**Issue by roninstar** [Azure-Samples/cognitive-services-speech-sdk#1165]
**Describe the bug**
We are using exactly the example provided for Keyword Recognition.  Once the wake word/ Key word is processed in the Recognized event it goes to SpeechEndDetected then SessionStopped.  This does not give our users the ability to say the wake word pause to see the device is listening then speak the phrase or question they want to ask our device.  

**To Reproduce**
Run this code and speak the key word and wait a moment before saying anything else then speak your sentence.  

**Expected behavior**
I would expect the Keyword Recognition to allow there to be a maximum of 5 seconds before stopping the session.

**Version of the Cognitive Services Speech SDK**
1.17 

**Platform, Operating System, and Programming Language**
 - OS: Android 8.1
 - Hardware - ARM 7
 - Programming language: C#
 -Xamarin.Andriod
Microsoft Visual Studio Enterprise 2019 
Version 16.10.2

**Additional context**
Create a Task then run the following code in the Task
 var config = SpeechConfig.FromSubscription(CognitiveServicesSettings.AzureSubscriptionKey, CognitiveServicesSettings.AzureRegionEastUs);
                
                kwsModelDir = spotterFilePath;
                var model = KeywordRecognitionModel.FromFile(kwsModelDir);

                // The phrase your keyword recognition model triggers on.
                var keyword = any keyword";

                _stopRecognition = new TaskCompletionSource<int>();
                var resultStr = "";
                byte channels = 1;
                byte bitsPerSample = 16;
                uint samplesPerSecond = 16000;
                var audioStreamFormat = AudioStreamFormat.GetWaveFormatPCM(samplesPerSecond, bitsPerSample, channels);

               _pushAudioInputStream = new PushAudioInputStream(audioStreamFormat);
               
                using (_audioInput = AudioConfig.FromStreamInput(_pushAudioInputStream))
                //using (_audioInput = AudioConfig.FromDefaultMicrophoneInput())
                {
                   
                    // Creates a speech recognizer using microphone as audio input.
                    using (var recognizer = new SpeechRecognizer(config, _audioInput))
                    {
                      
                        // Subscribes to events.
                        recognizer.Recognized += (s, e) =>
                        {
                            switch(e.Result.Reason)
                            {
                                case ResultReason.RecognizedKeyword:
                                
                                        resultStr = e.Result.Text;
                                        Console.WriteLine( "RECOGNIZED KEYWORD: " + resultStr);
                                   
                                    break;

                                case ResultReason.RecognizedSpeech:
                                    var resultStr = e.Result.Text;
                                    var parseSpeech = ParseWakeWord(e.Result.Text);
                                    if (!string.IsNullOrWhiteSpace(parseSpeech))
                                    {
                                        Console.WriteLine( $"RECOGNIZED: '{parseSpeech}'");

                                    }

                                    break;
                                case ResultReason.NoMatch:
                            
                                    
                                    resultStr = "NOMATCH: Speech could not be recognized.";
                                    

                                    break;
                                
                              default:
                                    Console.WriteLine(  e.Result.Reason);
                                        break;
                            }
                            

                        };

                        recognizer.Canceled += (s, e) =>
                        {

                            var cancellation = CancellationDetails.FromResult(e.Result);
                            resultStr = $"CANCELED: Reason={cancellation.Reason} ErrorDetails={cancellation.ErrorDetails}";
                     
                            if (cancellation.Reason == CancellationReason.Error)
                            {

                            }

                            _stopRecognition.TrySetResult(0);
                        };
                        
                        recognizer.SessionStarted += (s, e) =>
                        {
                        
                        };
                        recognizer.SpeechStartDetected += (sender, args) =>
                        {
                        };
                        recognizer.SpeechEndDetected += (s, e) =>
                        {

                        };
                        recognizer.SessionStopped += (s, e) =>
                        {
                            _stopRecognition.TrySetResult(0);
                        };

                        // Starts continuous recognition using the keyword model. Use StopKeywordRecognitionAsync() to stop recognition.
                         recognizer.StartKeywordRecognitionAsync(model).ConfigureAwait(false);
                        // Waits for a single successful keyword-triggered speech recognition (or error).
                        // Use Task.WaitAny to keep the task rooted.
                        Task.WaitAny(new[] { _stopRecognition.Task });
                       
                        recognizer.StopKeywordRecognitionAsync().ConfigureAwait(false);
 
                        _phraseSpotTask = null;
                        _pushAudioInputStream.Close();

                    }
                }

**Comment by roninstar** [comment]
Thanks for following up with me.  We look forward to your work around! best regards,
Mark

**Comment by Xiliath** [comment]
I am running into the same problem at the moment.
Is there a workaround or solution to this using SpeechRecognizer?

**Comment by chschrae** [comment]
A workaround you could do is to use the KeywordRecognizer independent of the SpeechRecognizer. This would allow you to start speech recognition only AFTER a keyword was spotted. If this doesn't work for you, please open a new issue.

**Comment by roninstar** [comment]
Starting the SpeechRecognizer after the KeywordRecognizer is recognized worked out as a good solution for us after all. Thank you
